cli/src/main.rs

```diff
@@ -67,35 +74,146 @@ pub fn help_command() {
+
+    if parts.is_empty() {
+        println!("No input received. Please try again");
+        return
+    }
+    let command = match Command::from_string(parts[0]) {
+        Some(cmd) => cmd,
+        None => {
+            println!("\x1b[1m\x1b[95m\nUnknown command: {}\x1b[0m", input.trim());
+            Command::Help
+        }
+    };
+
+    let arguments = if parts.len() > 1 {
+        Some(&parts[1..])
+    } else {
+        None
+    };
+    
+    match command {
+        Command::Start => todo!(),
+        Command::Create => {
+            create_room(stdin_tx);
+        },
+        Command::Join => join_room(arguments, stdin_tx),
+        Command::Send => {
+            send_message(arguments, stdin_tx);
+        },
+        Command::List => todo!(),
+        Command::Help => {
             help_command();
+        },
+        Command::Quit => {
+            println!("Goodbye!");
+            std::process::exit(0);
         }
-        input.clear();
     }
 }
+
+#[derive(Serialize, Deserialize)]
+struct FromServerMessage {
+    room: Option<String>,
+    message: Option<String>
+}
+
+fn handle_server_data(data: String) {
+    println!("data: {}", data);
+    // TODO: fix so server is sending right data in this form
+    // let server_data: FromServerMessage = serde_json::from_str(&data).unwrap();
+    println!("\x1b[1m\x1b[95m>>\x1b[0m");
+    println!("\x1b[1m\x1b[36m{:?}\x1b[0m", data);
+    println!("\x1b[1m\x1b[95m<<\x1b[0m");
+}
+
+#[derive(Serialize, Deserialize)]
+struct ToServerMessage {
+    // TODO: change this
+    message_type: Option<String>,
+    room_id: Option<String>,
+    secret_message: Option<String>,
+    key: Option<String>
+}
+
+fn create_room(stdin_tx: &UnboundedSender<WS_Message>) {
+    let msg = ToServerMessage {
+        message_type: Some("create".to_string()),
+        secret_message: None,
+        room_id: None,
+        key: None
+    };
+    let json_msg = serde_json::to_string(&msg).unwrap();
+    stdin_tx.unbounded_send(WS_Message::text(json_msg)).unwrap();
+}
+
+fn join_room(arguments: Option<&[&str]>, stdin_tx: &UnboundedSender<WS_Message>) {
+    match arguments {
+        Some(args) => {
+            let msg = ToServerMessage {
+                message_type: Some("join".to_string()),
+                secret_message: None,
+                // TODO: send the actual room id!
+                room_id: Some(args[0].to_string()),
+                key: None
+            };
+            let json_msg = serde_json::to_string(&msg).unwrap();
+            stdin_tx.unbounded_send(WS_Message::text(json_msg)).unwrap();
+        },
+        None => {
+            println!("missing arguments for join");
+        },
+    }
+}
+
+// TODO: cache the room so we don't need to use it as an argument (will need to use a lock ?)
+// right now do $ send <message> <room_id>
+// TODO: fix so we don't crash when user doesnt pass in the room id (or just fix so we dont need
+// to pass the room id
```

Comment:
yeah we can just store it in memory, since we dont plan to keep the  rooms active forever. -> once all users of a room disconnect, we kill the room on the server?